package/service/group_service.go

```diff
@@ -112,9 +166,71 @@ func (gs *groupService) GetGroup(tx *gorm.DB, groupId int64) (*schemas.Group, er
 		return nil, err
 	}
 
+	if current_user.Role == models.UserRoleTeacher.String() && group.CreatedBy != current_user.Id {
+		return nil, errors.ErrNotAuthorized
+	}
+
 	return gs.modelToSchema(group), nil
 }
 
+func (gs *groupService) AddUsersToGroup(tx *gorm.DB, current_user schemas.User, groupId int64, userIds []int64) error {
+	err := utils.ValidateUserRole(current_user.Role, []models.UserRole{models.UserRoleAdmin, models.UserRoleTeacher})
+	if err != nil {
+		return err
+	}
+
+	group, err := gs.groupRepository.GetGroup(tx, groupId)
+	if err != nil {
+		return err
+	}
+
+	if current_user.Role == models.UserRoleTeacher.String() && group.CreatedBy != current_user.Id {
+		return errors.ErrNotAuthorized
+	}
+
+	for _, userId := range userIds {
+		_, err := gs.userRepository.GetUser(tx, userId)
+		if err != nil {
+			return err
+		}
+
+		err = gs.groupRepository.AddUserToGroup(tx, groupId, userId)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (gs *groupService) GetGroupUsers(tx *gorm.DB, current_user schemas.User, groupId int64) ([]schemas.User, error) {
+	err := utils.ValidateUserRole(current_user.Role, []models.UserRole{models.UserRoleAdmin, models.UserRoleTeacher})
+	if err != nil {
+		return nil, err
+	}
+
+	group, err := gs.groupRepository.GetGroup(tx, groupId)
+	if err != nil {
+		return nil, err
+	}
+
+	if current_user.Role == models.UserRoleTeacher.String() && group.CreatedBy != current_user.Id {
```

Comment:
Can't we make User schema, role field have the same type as the model one? I don't like conversion to string during comparison.

We can define custom types in for example domains/types package